.github/CODEOWNERS

```diff
@@ -0,0 +1,87 @@
+#
+# - Comment lines begin with `#` character.
+# - Each line is a file pattern followed by one or more owners.
+# - The '*' pattern is global owners.
+# - Order is important. The last matching pattern has the most precedence.
+
+# Crates
+/crates/flake8_to_ruff/
+/crates/ruff/
+/crates/ruff_benchmark/
+/crates/ruff_cache/
+/crates/ruff_cli/
+/crates/ruff_dev/
+/crates/ruff_diagnostics/
+/crates/ruff_formatter/
+/crates/ruff_index/
+/crates/ruff_macros/
+/crates/ruff_python_ast/
+/crates/ruff_python_formatter/
+/crates/ruff_python_semantic/
+/crates/ruff_python_stdlib/
+/crates/ruff_python_whitespace/
+/crates/ruff_rustpython/
+/crates/ruff_testing_macros/
+/crates/ruff_textwrap/
+/crates/ruff_wasm/
+
+# Jupyter
+/crates/ruff/src/jupyter/ @dhruvmanila
+
+# Rules
+/crates/ruff/src/rules/airflow/
+/crates/ruff/src/rules/copyright/
+/crates/ruff/src/rules/eradicate/
+/crates/ruff/src/rules/flake8_2020/
+/crates/ruff/src/rules/flake8_annotations/
+/crates/ruff/src/rules/flake8_async/
+/crates/ruff/src/rules/flake8_bandit/
+/crates/ruff/src/rules/flake8_blind_except/
+/crates/ruff/src/rules/flake8_boolean_trap/
+/crates/ruff/src/rules/flake8_bugbear/
+/crates/ruff/src/rules/flake8_builtins/
+/crates/ruff/src/rules/flake8_commas/
+/crates/ruff/src/rules/flake8_comprehensions/
+/crates/ruff/src/rules/flake8_datetimez/
+/crates/ruff/src/rules/flake8_debugger/
+/crates/ruff/src/rules/flake8_django/
+/crates/ruff/src/rules/flake8_errmsg/
+/crates/ruff/src/rules/flake8_executable/
+/crates/ruff/src/rules/flake8_fixme/
+/crates/ruff/src/rules/flake8_future_annotations/
+/crates/ruff/src/rules/flake8_gettext/
+/crates/ruff/src/rules/flake8_implicit_str_concat/
+/crates/ruff/src/rules/flake8_import_conventions/
+/crates/ruff/src/rules/flake8_logging_format/
+/crates/ruff/src/rules/flake8_no_pep420/
+/crates/ruff/src/rules/flake8_pie/
+/crates/ruff/src/rules/flake8_print/
+/crates/ruff/src/rules/flake8_pyi/
+/crates/ruff/src/rules/flake8_pytest_style/
+/crates/ruff/src/rules/flake8_quotes/
+/crates/ruff/src/rules/flake8_raise/
+/crates/ruff/src/rules/flake8_return/
+/crates/ruff/src/rules/flake8_self/
+/crates/ruff/src/rules/flake8_simplify/
+/crates/ruff/src/rules/flake8_slots/
+/crates/ruff/src/rules/flake8_tidy_imports/
+/crates/ruff/src/rules/flake8_todos/
+/crates/ruff/src/rules/flake8_type_checking/
+/crates/ruff/src/rules/flake8_unused_arguments/
+/crates/ruff/src/rules/flake8_use_pathlib/
+/crates/ruff/src/rules/flynt/
+/crates/ruff/src/rules/isort/
+/crates/ruff/src/rules/mccabe/
+/crates/ruff/src/rules/numpy/
+/crates/ruff/src/rules/pandas_vet/
+/crates/ruff/src/rules/pep8_naming/
+/crates/ruff/src/rules/perflint/
+/crates/ruff/src/rules/pycodestyle/
+/crates/ruff/src/rules/pydocstyle/
+/crates/ruff/src/rules/pyflakes/
+/crates/ruff/src/rules/pygrep_hooks/
+/crates/ruff/src/rules/pylint/
+/crates/ruff/src/rules/pyupgrade/
+/crates/ruff/src/rules/ruff/
+/crates/ruff/src/rules/tryceratops/
```

Comment:
What's the process by which this is kept up-to-date? (E.g., if we add new directories, or rename, or remove?)

Reply:
I'm seeing a message on top of the file saying "This CODEOWNERS file is valid". I believe GitHub checks it against the repository. Let me do a quick commit and verify.

If not, we could use https://github.com/mszostok/codeowners-validator although it seems to be not maintained anymore.

Reply:
Hmm, I think GitHub seems to be only checking for syntax. Also, the `codeowners-validator` is giving me segmentation fault :(

Reply:
The path could contain patterns as well but if we don't use patterns then we could do a manual check on just the existence of path in CI.

Reply:
In that case, I might vote to only include the folders for which we're enforcing rules, and add them as we go. What do you think?

Reply:
Yeah, makes sense. Do you have any other folders in mind? (I'll comment out the rest before merging)

Reply:
Na, that's fine, I may add some later :)

Also: it may be easier just to remove rather than comment-out, since they'll probably get stale if we don't maintain them?